### PR TITLE
[VL] Support regr_sxy aggregate function

### PR DESCRIPTION
**Pull request by liujiayi771**
## What changes were proposed in this pull request?

Support `regr_sxy` aggregate function.

## How was this patch tested?

Add `VeloxAggregateFunctionsSuite.regr_sxy` test case.

### CODE REVIEW

**Comment by liujiayi771** [comment]
`regr_sxx` and `regr_syy` will be supported after the merge of https://github.com/facebookincubator/velox/pull/9360, and will be proposed in a separate PR.

File: backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala [hunk 1]
```diff
@@ -428,6 +428,25 @@ abstract class VeloxAggregateFunctionsSuite extends VeloxWholeStageTransformerSu
     }
   }
 
+  testWithSpecifiedSparkVersion("regr_sxy", Some("3.4")) {
```
Comment (yma11) [inline backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala:431]: Is this agg function not supported in other Spark version? Maybe you can consider using `isSparkVersionAtleast` to control the test coverage.

Comment (liujiayi771) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala:431]: Perhaps `isSparkVersionAtleast ` is same as `testWithSpecifiedSparkVersion`.

Comment (PHILO-HE) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala:431]: Yes, they are equivalent. `testWithSpecifiedSparkVersion` looks more clear to reader.

**Review by PHILO-HE — APPROVED** [review]
Thanks!

File: backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala (hunk 1 above)
Comment (yma11) [reply backends-velox/src/test/scala/org/apache/gluten/execution/VeloxAggregateFunctionsSuite.scala:431]: You guys are correct. Seems `isSparkVersionAtleast` is not necessary and I see there are some usage. Let me do a clean up as a follow up. Thanks.

**Review by yma11 — APPROVED** [review]

**Pull request merged**